public/app/core/services/backend_srv.ts

```diff
@@ -148,76 +159,17 @@ export class BackendSrv implements BackendService {
     const requestId = options.requestId ?? `chunked-${this.chunkRequestId++}`;
     const controller = new AbortController();
 
-    let url: string;
-    try {
-      url = parseUrlFromOptions(options);
-    } catch (error) {
-      return throwError(() => error);
-    }
-
     const init = parseInitFromOptions({
       ...options,
       requestId,
       abortSignal: controller.signal,
     });
 
     return new Observable((observer) => {
-      let done = false;
       // Calling fromFetch explicitly avoids the request queue
-      const sub = this.dependencies.fromFetch(url, init).subscribe({
-        next: (response) => {
-          const rsp = {
-            status: response.status,
-            statusText: response.statusText,
-            ok: response.ok,
-            headers: response.headers,
-            url: response.url,
-            type: response.type,
-            redirected: response.redirected,
-            config: options,
-            traceId: response.headers.get(GRAFANA_TRACEID_HEADER) ?? undefined,
-            data: undefined,
-          };
-
-          if (!response.body) {
-            observer.next(rsp);
-            observer.complete();
-            return;
-          }
-
-          const reader = response.body.getReader();
-          async function process() {
-            while (reader && !done) {
-              if (controller.signal.aborted) {
-                reader.cancel(controller.signal.reason);
-                console.log(requestId, 'signal.aborted');
-                return;
-              }
```

Comment:
When I added unit tests for the `chunked` function I couldn't reach this code. Will need an extra pair of eyes on the replacement in the new function `getChunkedResponseObserver` (lines in comment below). Ideally if someone could help me test this on prem and on cloud that would be great, thank you 🙏 

```ts
// Setup onabort callback so that we can cancel the reader properly
controller.signal.onabort = () => {
  reader.cancel(controller.signal.reason);
  console.log(requestId, 'signal.aborted');
};
```